h2/src/main/org/h2/util/ValueHashMap.java

```diff
@@ -174,6 +191,42 @@ public V get(Value key) {
         }
         return list;
     }
+    
+    public Iterable<Map.Entry<Value, V>> entries() {
+        return new EntryIterable();
+    }
+    
+    private final class EntryIterable implements Iterable<Map.Entry<Value, V>> {
+        @Override
+        public Iterator<Map.Entry<Value, V>> iterator() {
+            return new EntryIterator();
+        }
+    }
+
+    private final class EntryIterator implements Iterator<Map.Entry<Value, V>> {
+        private int keysIndex = -1;
+        private int left = size;
+
+        @Override
+        public boolean hasNext() {
+            return left > 0;
+        }
+
+        @Override
+        public Map.Entry<Value, V> next() {
+            left--;
+            do {
+                keysIndex++;
+                if (keys[keysIndex] != null && keys[keysIndex] != ValueNull.DELETED)
+                    return new DataUtils.MapEntry<Value, V>(keys[keysIndex], values[keysIndex]);
+            } while (true);
```

Comment:
From my point of view `do while (true)` is a very strange way to write an infinite loop. `for(;;)` is better. Or even `while(true)` is better. It does not really matter, however.

Also it looks like we can read key from array only once per loop to reduce size of bytecode and improve readability:
```Java
                Value key = keys[keysIndex];
                if (key != null && key != ValueNull.DELETED)
                    return new DataUtils.MapEntry<>(key, values[keysIndex]);
```

Reply:
Class `DataUtils.MapEntry` can be removed and replaced anywhere with `java.util.AbstractMap.SimpleImmutableEntry` that is available since Java 6 / Android API 9.